gax/src/main/java/com/google/api/gax/rpc/BidiStreamingCallable.java

```diff
@@ -44,22 +44,196 @@
 
   protected BidiStreamingCallable() {}
 
+  /**
+   * The "base" method from which other forms of {@code call}s are derived. Most users will not need
+   * to call this method directly.
+   *
+   * <p>However, it is {@code public}, since library authors might want to call this method in
+   * adaptor classes.
+   */
+  public abstract ClientStream<RequestT> internalCall(
+      ResponseObserver<ResponseT> responseObserver,
+      ClientStreamReadyObserver<RequestT> onReady,
+      ApiCallContext context);
+
+  /**
+   * Listens to server responses and send requests when the network is free. Example usage:
+   *
+   * <pre>{@code
+   * final Iterator<Integer> sourceDataIterator = intCollection.iterator();
+   * BidiStreamObserver<Integer, String> bidiStreamObserver = new BidiStreamObserver<Integer, String>() {
+   *   public void onStart(StreamController controller) {
+   *     // no-op
+   *   }
+   *
+   *   public void onResponse(String response) {
+   *     System.out.println(response);
+   *   }
+   *
+   *   public void onComplete() {
+   *     System.out.println("done!");
+   *   }
+   *
+   *   public void onError(Throwable t) {
+   *     System.out.println("error: " + t);
+   *   }
+   *
+   *   public void onReady(ClientStream<Integer> stream) {
+   *     while (sourceDataIterator.hasNext()) {
+   *       if (stream.isReady()) {
+   *         stream.send(sourceDataIterator.next());
+   *       } else {
+   *         // It's OK we haven't consumed the whole iterator;
+   *         // onReady will be called again when the network becomes free.
+   *         return;
+   *       }
+   *     }
+   *     // We ran out of things to send.
+   *     stream.close();
+   *   }
+   * };
+   *
+   * bidiStreamingCallable.bidiCall(bidiStreamObserver);
+   * }</pre>
+   */
+  public void bidiCall(final BidiStreamObserver<RequestT, ResponseT> bidiObserver) {
```

Comment:
I had in mind reverting this name to `call` since the ambiguity with the other function is resolved, due to it now being named `splitCall`